RiotSwiftUI/Modules/Room/RoomUpgrade/Service/MatrixSDK/RoomUpgradeService.swift

```diff
@@ -37,11 +37,20 @@ class RoomUpgradeService: RoomUpgradeServiceProtocol {
     private(set) var errorSubject: CurrentValueSubject<Error?, Never>
     private(set) var currentRoomId: String
 
+    var parentSpaceName: String? {
+        guard let parentId = self.parentId, let parent = session.spaceService?.getSpace(withId: parentId) else {
```

Comment:
There are a lot of optionals here that could result in nil name, should we log any of them as failures? E.g. it is clear that name should be nil if there is no identifier, but is it a valid case that `getSpace(withId: parentId)` returns nil?

Reply:
The `parentSpaceId` can be `nil` as the room upgrade can be triggered from the `home` space. I added a log for the second one but this most likely not happen.